config/promitor/scraper/metrics.yaml

```diff
@@ -185,6 +185,20 @@ metrics:
         type: Average
     resourceDiscoveryGroups:
       - name: front-door-landscape
+  - name: promitor_demo_frontdoor_backend_health_per_backend_pool_and_backend
+    description: "Health percentage for a backend in Azure Front Door"
+    resourceType: FrontDoor
+    labels:
+      app: promitor
+    azureMetricConfiguration:
+      metricName: BackendHealthPercentage
+      dimensions:
+        - name: BackendPool
+        - name: Backend
```

Comment:
We should verify that the integration tests checks if all the dimensions are there imo

Reply:
Can you point me to where checks like this are / would need to be added in the source code?

Reply:
You can find all related tests in https://github.com/tomkerkhove/promitor/tree/master/src/Promitor.Tests.Integration/Services/Scraper

This test verifies that all configured metrics are reported:
https://github.com/tomkerkhove/promitor/blob/master/src/Promitor.Tests.Integration/Services/Scraper/MetricSinks/PrometheusScrapeMetricSinkTests.cs#L36

We can use this as inspiration to see if we can pull in the dimension information; but a dedicated test scenario for verifying this would be idea.

It's OK to use one from `config/promitor/scraper/metrics.yaml` (ie `promitor_demo_frontdoor_backend_health_per_backend_pool_and_backend`) and assume it's there so you can just assert on the two dimensions. No need to dynamically parse and get them to keep it simple, unless you want to.

While doing that, let's do the same for one that is using the old dimension field as well.

Reply:
I've added the test `Prometheus_Scrape_ExpectedDimensionsAreAvailable` for this now, but it seems to cause the integration tests to just run forever with a lot more of the `Current metric type is not a gauge, ignoring` messages than previously. I can't find what causes these messages in general however, nor why the test would be causing so many more of them and apparently some kind of infinite loop. Do you know what could be wrong with the current version of the new test?

On a related note, it's not really possible for me to run the integration tests locally somehow instead of relying on the Azure Pipelines, right?

Reply:
So the test basically starts polling Promitor to check if the metric is available or not. If it is not, it waits and tries again. See https://github.com/tomkerkhove/promitor/pull/2123/files#diff-b28d4d780cd76e1f4acd8507d02936223e16a5fc331c2f52104add9ec27d6603R105

Most likely the metric was not available and it kept waiting, however it should not wait forever. Do you still know what build it was?

You should be able to run it locally though: https://github.com/tomkerkhove/promitor/blob/master/CONTRIBUTING.md#running-promitor

Reply:
Yes, it might be related. Can you nskip that please so we have another run?

Reply:
Sure, I did that now. Last time it hadn't stopped after 20 minutes or so though, so you might have to cancel it manually at one point